### ISSUE TITLE
addDefault doesn't work since 1.8 if file contains only defaults

### ISSUE THREAD

**Issue by ChimpGamer** [Carleslc/Simple-YAML#64]
I noticed that since the 1.8 update that addDefault doesn't work any more since the test.yml is empty after being created and saved.
I was able to recreate this in a test project which I uploaded to github: https://github.com/ChimpGamer/SimpleYamlBug/blob/main/src/main/kotlin/nl/chimpgamer/simpleyamlbug/Main.kt

Am I missing something or is it really a bug?

**Comment by Carleslc** [comment]
Yes, it is a bug of version 1.8.

It happens only when there are no keys set (only defaults, without any use of [`set`](https://carleslc.me/Simple-YAML/doc/Simple-Yaml/org/simpleyaml/configuration/file/YamlFile.html#set-java.lang.String-java.lang.Object-)  method).

See the following example. I added a set call to your example.

```kotlin
val yamlFile = YamlFile("test.yml")

yamlFile.createOrLoad()

yamlFile.addDefault("default", "test")

yamlFile.set("key", 0)

yamlFile.save()
```

```yaml
default: test
key: 0
```

**Comment by Carleslc** [comment]
Reopen until the 1.8.1 release

**Comment by Carleslc** [comment]
Workaround for 1.8:

```kotlin
fun addDefault(yamlFile: YamlFile, path: String, value: Any) {
    if (!yamlFile.isSet(path)) {
        yamlFile.set(path, value);
    }
    yamlFile.addDefault(path, value);
}

fun main() {
    val yamlFile = YamlFile("examples/test64.yml")

    yamlFile.createOrLoad()

    addDefault(yamlFile, "default", "test")

    yamlFile.save()
}
```

```yaml
default: test
```

**Comment by Carleslc** [comment]
Fixed in [1.8.1](https://github.com/Carleslc/Simple-YAML/releases/tag/1.8.1)
